e2e/files.spec.ts

```diff
@@ -0,0 +1,283 @@
+  ).not.toBeVisible()
+
+  await confirmAndClosePermissionsSavedAlert(page)
+}
+
+let userProject: Project
+let fileHandleIds: string[] = []
+
+test.describe('Files', () => {
+  testAuth.beforeAll(async ({ browser, storageStatePaths }) => {
+    const userContext = await browser.newContext({
+      storageState: storageStatePaths['swc-e2e-user'],
+    })
+    const userPage = await userContext.newPage()
+    const userAccessToken = await getAccessTokenFromCookie(userPage)
+
+    // create project
+    const userProjectName = generateEntityName('project')
+    const userProjectId = await createProject(
+      userProjectName,
+      userAccessToken,
+      userPage,
+    )
+    userProject = { name: userProjectName, id: userProjectId }
+
+    await userContext.close()
+  })
+
+  testAuth.afterAll(async ({ browser }) => {
+    const context = await browser.newContext()
+    const page = await context.newPage()
+    const accessToken = getAdminPAT()
+
+    // delete project
+    if (userProject.id) {
+      await deleteProject(userProject.id, accessToken, page)
+    }
+
+    // delete fileHandles
+    for await (const fileHandleId of fileHandleIds) {
+      await deleteFileHandleWithRetry(accessToken, fileHandleId, page)
+    }
+
+    await context.close()
+  })
+
+  testAuth(
+    'File sharing',
+    async ({ userPage, validatedUserPage, browserName }, testInfo) => {
+      if (browserName === 'webkit') {
+        test.info().annotations.push({
+          type: 'very slow',
+          description: `webkit in CI only. May be related to the InviteWidget. 
+            Evaluate if timeout can be removed after addressing 
+            https://sagebionetworks.jira.com/browse/SWC-6569.`,
+        })
+        test.setTimeout(testInfo.timeout * 5)
+      }
+
+      const userName = userConfigs['swc-e2e-user'].username
+      const validatedUserName = userConfigs['swc-e2e-user-validated'].username
+
+      const { fileEntityId, fileName } = await testAuth.step(
+        'Create a file',
+        async () => {
+          const userAccessToken = await getAccessTokenFromCookie(userPage)
+          const fileName = `${generateEntityName('file')}.txt`
+          const { fileEntityId, fileHandleId } = await createFile(
+            fileName,
+            'Created for SWC e2e testing.',
+            'text/txt',
+            userProject.id,
+            userAccessToken,
+            userPage,
+          )
+          fileHandleIds.push(fileHandleId)
+          return { fileEntityId, fileName }
+        },
+      )
+
+      await testAuth.step('First user can view a private file', async () => {
+        await userPage.goto(`${entityUrlPathname(userProject.id)}/files`)
+        await waitForInitialPageLoad(userPage)
+
+        const fileLink = userPage.getByRole('link', { name: fileName })
+        await expect(fileLink).toBeVisible()
+
+        await fileLink.click()
+        await expect(
+          userPage.getByText(`Discussion about ${fileName}`),
```

Comment:
Just curious, why look for "Discussion about" ?

Reply:
Ah, good question -- this was an easy locator to confirm that the File page had loaded, but I'll add some other expectations about the File page for completeness!